assets/app/scripts/controllers/create.js

```diff
@@ -94,6 +107,213 @@ angular.module('openshiftConsole')
       $scope.filter.tag = tag;
     };
 
+    $scope.aceLoaded = function(editor) {
+      $scope.editor.session = editor.getSession();
+      $scope.editor.session.setOption('tabSize', 2);
+      $scope.editor.session.setOption('useSoftTabs', true);
+      editor.setDragDelay = 0;
+      editor.$blockScrolling = Infinity;
+
+      $('.from-file .editor').animate({
+        height: Math.floor(window.innerHeight * 0.50) + 'px'
+      }, 30, function() {
+        editor.resize();
+      });
+    };
+
+    // Check if the editor isn't empty to disable the 'Add' button. Also check in what format the input is in (JSON/YAML) and change
+    // the editor accordingly.
+    $scope.aceChanged = function() {
+      $scope.editor.empty = false;
+      if (!$scope.editor.content) {
+        $scope.editor.empty = true;
+      }
+      try {
+        JSON.parse($scope.editor.content);
+        $scope.editor.session.setMode("ace/mode/json");
+      } catch (e) {
+        try {
+          YAML.parse($scope.editor.content);
+          $scope.editor.session.setMode("ace/mode/yaml");
+        } catch (e) {
+          return;
+        }
+      }
+    };
+
+    $scope.create = function() {
+      delete $scope.alerts['create'];
+      delete $scope.alerts['parsing'];
+      var resource;
+
+      // Trying to auto-detect what format the input is in. Since parsing JSON throws only SyntexError
+      // exception if the string to parse is not valid JSON, it is tried first and then the YAML parser
+      // is trying to parse the string. If that fails it will print the reason. In case the real reason
+      // is JSON related the printed reason will be "Reason: Unable to parse", in case of YAML related
+      // reason the true reason will be printed, since YAML parser throws an error object with needed
+      // data.
+      try {
+        resource = JSON.parse($scope.editor.content);
+      } catch (e) {
+        try {
+          resource = YAML.parse($scope.editor.content);
+        } catch (e) {
+          $scope.alerts['parsing'] = {
+            type: "error",
+            message: "An error occurred during parsing editors content.",
+            details: "Reason: " + e.message
+          };
+          return;
+        }
+      }
+
+      $scope.resourceKind = resource.kind;
+      
+      if ($scope.resourceKind.endsWith("List")) {
+        $scope.resourceList = resource.items;
+      } else {
+        $scope.resourceList = [resource];
+        $scope.resourceName = resource.metadata.name;
+      }
+
+      $scope.updateResources = [];
+      $scope.createResources = [];
+
+      var itemsProcessed = 0;
+      angular.forEach($scope.resourceList, function(item) {
+        var itemKind = item.kind;
```

Comment:
Well I though about adding and error but I wasn't sure if we wanna have the validation done client side. Then we would have to check for all the fields that have to be filled for each resources kind. Do we want it ?